src/map.c

```diff
@@ -664,6 +679,22 @@ int wally_map_keypath_add(struct wally_map *map_in,
     return ret;
 }
 
+int wally_map_merkle_path_add(struct wally_map *map_in,
+                          const unsigned char *pub_key, size_t pub_key_len,
+                          const unsigned char *merkle_hashes, size_t merkle_hashes_len)
+{
+
+    if (!map_in || pub_key_len != EC_XONLY_PUBLIC_KEY_LEN)
+        return WALLY_EINVAL;
+
+    if (map_in->verify_fn != wally_merkle_path_xonly_public_key_verify)
+        return WALLY_EINVAL; /* Not a taproot keypath map */
+
+    /* Add map for tap leaves */
+    return map_add(map_in, pub_key, pub_key_len,
+                  merkle_hashes_len ? merkle_hashes : NULL, merkle_hashes_len, false, false);
+}
```

Comment:
formatting! also please keep line lengths to less than 100 if possible.

Reply:
>merkle_hashes_len ? merkle_hashes : NULL

Its always an error in wally to pass a valid ptr with 0 length or a null ptr with a length. Please check this using `BYTES_INVALID` above and remove the ternary.